[DOCS] Clarifies encrypting communications in Kibana

lcawl opened the pull request:
The Kibana User Guide contains information about encrypting communications in step 5 of the "Configuring Security in Kibana" page: https://www.elastic.co/guide/en/kibana/master/using-kibana-with-security.html

I found similar information in the "Using Kibana in production" page: https://www.elastic.co/guide/en/kibana/master/production.html#enabling-ssl

This PR merges the information from those two sources and attempts to address some places where additional clarification is required.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/3351/)

lcawl commented:
@epixa Do you think it will be less confusing when the "Configuring" security section is moved under the "Set Up Kibana" section (i.e. no longer under "Set Up X-Pack") in https://github.com/elastic/kibana/pull/19182 ?

Or do we need to add a note in step 1 to clearly indicate that X-Pack Security is not required for that step?

epixa commented:
I think that would be slightly less confusing, but it's still giving the impression that TLS in Kibana is a paid feature, when it isn't. As it stands now, the entire "encrypting communications with kibana" has nothing at all to do with x-pack, but you can only access it if you first navigate into setting up x-pack and then to this new page. With #19182, the situation is slightly better, but you still need to navigate from "general setup stuff" to "paid security features" in order to get to "free tls stuff". If we're super clear about what is paid and what is not, this might be OK.

epixa commented:
cc @elastic/kibana-security

lcawl commented:
Thanks, @epixa, I've added a note to the first step in x-pack/docs/en/security/securing-communications/index.asciidoc and removed the "xpack" icon from the title of this page.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/3436/)

epixa commented:
I don't think this is sufficient, but I don't think the content on the page itself is the problem. With this change, there's just no reasonable expectation that a person will be able to find this documentation anymore unless they are paying customers that use security.  This is where this documentation now lives in the tree:

![screen shot 2018-05-18 at 12 09 46 pm](https://user-images.githubusercontent.com/53814/40245770-858272e8-5a94-11e8-9026-66f77adc29f3.png)

Perhaps we should put this on hold until #19182 is sorted?

lcawl commented:
I've merged https://github.com/elastic/kibana/pull/19182 and rebased this PR.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic-kibana-pull-request/3485/)
